react/daily-art-prompt/src/store/rootReducer.js

```diff
@@ -1,186 +1,149 @@
+          const updatedComments = image.comments.map(comment => {
+            if (comment.id === action.payload.commentId) {
+              return {
+                ...comment,
+                deleted: true
+              };
+            } else {
+              return comment;
             }
-            else {
-                return state
-            }
-        case TYPES.UPDATE_PREVIOUS_INDEX:
-            const updatedIndex = action.payload.index - 1;
-            if (updatedIndex >= 0) {
-                return updatedIndex
+          });
+          return {
+            ...image,
+            comments: updatedComments
+          };
+        } else {
+          return image;
+        }
+      });
+
+    case TYPES.UPDATE_COMMENT_EDITING:
+      return state.map(image => {
+        if (image.id === action.payload.imageId) {
+          const updatedCommentsClassName = image.comments.map(comment => {
+            if (comment.id === action.payload.commentId) {
+              return {
+                ...comment,
+                editing: !comment.editing
+              };
+            } else {
+              return comment;
             }
-            else {
-                return state
+          });
+          return {
+            ...image,
+            comments: updatedCommentsClassName
+          };
+        } else {
+          return image;
+        }
+      });
+
+    case TYPES.EDIT_COMMENT:
+      return state.map(image => {
+        if (image.id === action.payload.imageId) {
+          const updatedComments = image.comments.map(comment => {
+            if (comment.id === action.payload.commentId) {
+              return {
+                ...comment,
+                text: action.payload.value,
+                editing: false
+              };
+            } else {
+              return comment;
             }
-        default:
-            return state
-    }
+          });
+          return {
+            ...image,
+            comments: updatedComments
+          };
+        } else {
+          return image;
+        }
+      });
 
-}
+    default:
+      return state;
+  }
+};
 
-export const promptsReducer = (state = [], action) => {
-    switch (action.type) {
-        case TYPES.SET_INITIAL_PROMPTS:
-            const date = new Date()
-            const prompt = {
-                id: state.length > 0
-                    ? state[state.length - 1].id + 1
-                    : 1,
-                date: `${date.getMonth()}-${date.getDate()}-${date.getFullYear()}`,
-                text: action.payload.text
-            }
-            return [...state, prompt]
-        default:
-            return state
-    }
-}
+export const dateReducer = (state = null, action) => {
+  switch (action.type) {
+    case TYPES.SET_INITIAL_DATE:
+      return moment().format("YYYY-MM-DD");
+    case TYPES.UPDATE_NEXT_DATE:
+      return action.payload.date;
+    case TYPES.UPDATE_PREVIOUS_DATE:
```

Comment:
both of these do the same thing. So either:

1. Make a single action 

or

2. Put both cases together:

```js
case TYPES.UPDATE_NEXT_DATE:
case TYPES.UPDATE_PREVIOUS_DATE:
      return action.payload.date;
```

Reply:
the issue has been taken care of